packages/blocks/src/api/utils.js

```diff
@@ -38,7 +38,15 @@ export function isUnmodifiedDefaultBlock( block ) {
 		return false;
 	}
 
-	// Cache a newly created default block.
+	// Invalidate default block cache if default block name changed.
+	if (
+		isUnmodifiedDefaultBlock.block &&
+		isUnmodifiedDefaultBlock.block.name !== defaultBlockName
```

Comment:
This line could be added to the next `if` statement and avoid the delete entirely.

Reply:
True :)